classes/4.html

```diff
@@ -56,16 +56,368 @@ <h3>¿Qué es React?</h3>
         <!-- JSX  -->
         <section>
           <h3>JSX</h3>
+          <section>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              const elemento = <h1>Hola, mundo!</h1>;
+            </code></pre>
+            <p>
+              Esta sintaxis de etiquetas no es ni un string ni HTML. Se llama JSX, y es una extensión de la sintaxis de JavaScript.
+              Si bien no es requerido su uso con React, se recomienda hacerlo para describir cómo debería ser la interfaz de usuario.
+            </p>
+          </section>
+          <section>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              const nombre = 'Juan Perez';
+              const elemento = <h1>Hola, {nombre}</h1>;
+            </code></pre>
+            <p>
+              Se puede poner cualquier expresión JavaScript dentro de llaves en JSX.
+              Por ejemplo, <b>2 + 2</b>, <b>usuario.nombre</b>, o <b>formatearNombre(usuario)</b>.
+            </p>
+          </section>
+          <section>
+            <p>
+              Puede usarse dentro de declaraciones <b>if</b>, loops, asignarlo a variables, aceptarlo como argumento, y retornarlo desde dentro de funciones:
+            </p>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              function obtenerSaludo(usuario) {
+                if (usuario) {
+                  return <h1>Hola, {formatearNombre(usuario)}!</h1>;
+                }
+                return <h1>Hola, Extraño.</h1>;
+              }
+            </code></pre>
+          </section>
+          <section>
+            <p>
+              Se pueden usar comillas para especificar strings literales como atributos:
+            </p>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              const elemento = <div tabIndex="0"></div>;
+            </code></pre>
+            <p>
+              También se pueden usar llaves para insertar una expresión JavaScript en un atributo:
+            </p>
+            <pre><code data-trim data-line-numbers class="lang-javascript">
+              const elemento = <img src={usuario.urlAvatar}></img>;
```

Comment:
El ejemplo está barbaro y es bueno, pero `reveal` te agrega comilla alrededor del src, dejandolo como `src="{usuario.urlAvatar}"` y mostrando algo contrario a la intención del código. 

Una forma de evitarlo es wrappear el código con script tags con `type="text/template"`. [Acá](https://revealjs.com/code) hay un poco más de info. Esto aplica o otros ejemplos también, y por lo visto también jode los self closing tags

```suggestion
              <script type="text/template">
                const elemento = <img src={usuario.urlAvatar}></img>;
              </script>
```

Reply:
Buenísimo! Eso era algo que me quedó pendiente para cambiar sí. Muchas gracias!

Reply:
Pronto!

Reply:
Genial 👍🏼 
Se me complicó verlo por el force update, obliga un poco a hacer la revisión de nuevo. En general está bueno evitarlos.
Me llama la atención tanto cambio en el `package-lock.json` sin tocar el `package.json`. Será por una versión distinta de npm? 🤔

Reply:
Sí, lo mismo pensé, pero no me había fijado que había cambiado todo eso. Lo revertí. Ahora sólo hay un archivo cambiado, el de la clase